### PR TITLE
ModLoader with tests

### PR DESCRIPTION
**Pull request by IhateTrains**

### CODE REVIEW

**Review by Idhrendur — APPROVED** [review]
Looks good to me, but please wait for Zemurin's approval as he's more familiar with this.

**Review by Zemurin — APPROVED** [review]
This is an ad verbatim transliteration of existing ModLoader with the notable difference of using FastZip, and the result of it properly bailing when fed crap, unlike ZipLib which gleefully unpacks a 50-byte .txt files into nothing and reports no errors.

Looks good.

**Pull request merged**
